src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object [DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            increaseCapacity();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index is incorect");
+        }
+        if (elements.length == size) {
+            increaseCapacity();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        while (list.size() + size >= elements.length) {
+            increaseCapacity();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elements[size++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validationOfIndex(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validationOfIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validationOfIndex(index);
+        T removeItem = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return removeItem;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        return remove(getIndex(element));
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void increaseCapacity() {
+        T[] old = elements;
+        int capacity = old.length + (old.length >> 1);
+        elements = (T[]) new Object[capacity];
+        System.arraycopy(old, 0, elements, 0, size);
+    }
+
+    private void validationOfIndex(int index) {
+        if (index < 0 || index >= size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds of ArrayList");
```

Comment:
It means adding **index** and **size** values to the msg
![image](https://user-images.githubusercontent.com/20326858/218448988-fcb74a05-9c61-466a-bd7c-26a3a17cb27b.png)

Reply:
Ok,thank you for the informative review